 crazys-wm-mod/CSurface.cpp

```diff
@@ -264,53 +264,39 @@ bool CSurface::DrawSurface(int x, int y, SDL_Surface* destination, SDL_Rect* cli
 	{
 		if (m_Temp == 0)
 		{
-			if (maintainRatio == true)
+			if (clip->w != m_Surface->w && clip->h != m_Surface->h)
 			{
-				if (clip->w != m_Surface->w && clip->h != m_Surface->h)
-				{
-					if (m_Surface->w > m_Surface->h)	// if the width is larger so scale down based on the width but keep aspect ratio
-						scaleX = scaleY = ((double)clip->w / (double)m_Surface->w);
-					else	// assume the height is larger
-						scaleX = scaleY = ((double)clip->h / (double)m_Surface->h);
-					m_Temp = zoomSurface(m_Surface, scaleX, scaleY, 1);
-				}
-			}

```

Comment:
You have two possible return failures, the ifs need else statements or they may not do anything.